### DISCUSSION TITLE
eglot grammarly support

### DISCUSSION THREAD

**Discussion by HumHongeKamyaab** [joaotavora/eglot#693] [Q&A] [answered]
will eglot officially support eglot-grammarly?

**Comment by joaotavora** [comment, marked as answer]
I don't know grammarly, but if it's a LSP server, it should be supported without the need for this extra package, which seems like a reasonably useless piece of elisp to me.

https://github.com/emacs-grammarly/eglot-grammarly/blob/master/eglot-grammarly.el

Apparently you just need:
```
(add-to-list 'eglot-server-programs
             ((text-mode markdown-mode whatever-mode) . "@emacs-grammarly/unofficial-grammarly-language-server"))
```

And then you need to create a `.dir-locals` according to the README in https://github.com/emacs-grammarly/eglot-grammarly

My advice is not to follow *any* other advice in that README like that `use-package` line with `call-interactively`.
